server.go

```diff
@@ -94,7 +93,15 @@ func NewServer(opts ...ServerOption) (*Server, error) {
 		temporal.WithClaimMapper(func(cfg *config.Config) authorization.ClaimMapper {
 			return claimMapper
 		}),
-		temporal.WithDynamicConfigClient(dynamicconfig.NewNoopClient()),
+	}
+
+	if len(c.DynamicConfig) > 0 {
+		// To prevent having to code fall-through semantics right now, we currently
+		// eagerly fail if dynamic config is being configured in two ways
+		if cfg.DynamicConfigClient != nil {
```

Comment:
Technically I could build a fall back here, but there wasn't enough value (our NoopClient from before basically ensured nobody used this in their config anyways).